doc/lang.md

```diff
@@ -1260,8 +1260,10 @@ field(r, "fld") // r is an identifier, "fld" is a string literal
 r.fieldNames()
 fieldNames(r)
 // record update: [r EXCEPT !.f = e]
-r.with(f, e)
-with(r, f, e)
+r.with("f", e)
+with(r, "f", e)
```

Comment:
Shall we also hide this from users? It has the same problem as `field`: it is not a typeable operator, because we have no type for record labels. In fact it was in an attempt to supply record field names by reference that Lasaro reported the unecpected behavior reported in #816.

Reply:
We can remove it from the manual. However, it's important to know about this operator, as the spread operator is immediately translated into `with`. So it may happen that the users would see a message that mentions `with`, e.g., in REPL.

Reply:
> It has the same problem as field: it is not a typeable operator, because we have no type for record labels. In fact it was in an attempt to supply record field names by reference that Lasaro reported the unecpected behavior reported in https://github.com/informalsystems/quint/issues/816.

I think these are two separate questions:
 - 1. Can we specify the generic type of `field` and `with` in our type system?
 - 2. Can we specify a concrete type of `field` and `with` when they are called with string literals?

The answer to the first question is "no". The answer to the second question is "yes". In principle, I don't see how it would be different from the case of `Set(...)`, which requires varargs. There are a few special operators, and we are trying to keep their number to minimum. Perhaps, it's an issue of documentation.

Reply:
The problem is that the type we give it is incorrect: it does not take a value of type string (as the field argument). It only takes a string literal. This means users may try to use value of type string to access a field value, which should be correct according to the typing given, but instead we error out. The problem I see is not that type is not generic, but that the type is incorrect: the operator does not work over
Values of the string.

Reply:
Do you mean the type in the documentation or elsewhere?